pkgs/development/ocaml-modules/mirage-xen/pkg-config.patch

```diff
@@ -0,0 +1,12 @@
+diff --git a/lib/bindings/mirage-xen.pc b/lib/bindings/mirage-xen.pc
+index a03a217..824c379 100644
+--- a/lib/bindings/mirage-xen.pc
++++ b/lib/bindings/mirage-xen.pc
+@@ -1,6 +1,4 @@
+-prefix=${pcfiledir}/../..
+-exec_prefix=${prefix}
+-libdir=${exec_prefix}/lib
++libdir=${pcfiledir}/../
```

Comment:
Hmm normally we keep `prefix` and friends? And prefix is set to `$out` by default as a best effort.

I went into the rabbit hole a bit, looking at https://github.com/ocaml/dune/issues/1253 and linked issues like:

 - https://github.com/mirage/mirage-solo5/pull/61
 - https://github.com/ocaml/dune/issues/680
 - https://github.com/ocaml/dune/pull/3104

It looks like perhaps the pieces are there and those issues can be resolved?

Reply:
Oh right, I completely forgot about that. I quickfixed this on a local branch and was meaning to follow up. I'll see what this is all about :)

Reply:
The issue boils down to the following: The pkg config file in the repository has been manually created and assumes the default relation between dune's prefix and libdir (which would be `"${prefix}/lib"` for the latter). However for `ocamlPackages` we install libraries into a version specific site-lib, so that doesn't work out.

This is probably fixable in `mirage-xen`'s build system, but they probably have little interest in fixing that before the upcoming rework.

Reply:
Should be better now!